jsonschema/tests/test_validators.py

```diff
@@ -771,6 +772,28 @@ class TestRefResolver(unittest.TestCase):
     stored_uri = "foo://stored"
     stored_schema = {"stored" : "schema"}
 
+    def prev_resolve_fragment(self, document, fragment):
+        fragment = fragment.lstrip(u"/")
```

Comment:
This seems like quite a lot of not-testing-specific-looking code making an appearance in a test. I don't have a specific recommendation yet on how to remove it, but it doesn't look right to me, especially not in combination with being patched into place (I'd like to sometime soon remove the `mock` dependency entirely, but alas).